packages/codemod/transformations/core/v2-to-v3/steps-component-migration.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  wrap,
+  getCoreImportsForFile,
+  getComponentNameOrAliasFromImports,
+  findComponentElements,
+  removeProp,
+  isPropExists,
+  addNewProp,
+  getPropValue,
+  findProps
+} from "../../../src/utils";
+import { TransformationContext } from "../../../types";
+
+/**
+ * 1. remove 'isOnPrimary' prop if exists and is true
+ * 2. add color={Steps.colors.PRIMARY} if 'isOnPrimary' exists and is true
+ */
+function transform({ j, root }: TransformationContext) {
+  const imports = getCoreImportsForFile(root);
+  const componentName = getComponentNameOrAliasFromImports(j, imports, "Steps");
+  if (!componentName) return;
+
+  const elements = findComponentElements(root, componentName);
+  if (!elements.length) return;
+
+  elements.forEach(elementPath => {
+    if (isPropExists(j, elementPath, "isOnPrimary")) {
+      const prop = findProps(j, elementPath, "isOnPrimary").get(0);
+      const propValue = getPropValue(j, prop.node);
+      if (propValue) {
```

Comment:
maybe be more explicit here
```if propValue === true```? in case for some reason this prop isn't a boolean
or is an overkill, and if someone misused it, it's their problem